File: src/main/kotlin/com/hyungu/whoseturn/controller/UserController.java
package com.hyungu.whoseturn.controller;

import com.hyungu.whoseturn.model.UserModel;
import com.hyungu.whoseturn.service.UserService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/users")
public class UserController {

    @Autowired
    private UserService userService;

    @PostMapping
    public UserModel addMember(@RequestBody UserModel userModel) {
        System.out.println("run PostMapping ++++++++++++");
        return userService.saveMember(userModel);
    }

    @GetMapping
    public List<UserModel> getAllMembers() {
        System.out.println("run PostMapping ++++++++++++");
        return userService.getAllMembers();
    }

    @GetMapping("/{id}")
    public UserModel getMemberById(@PathVariable Long id) {
        return userService.getMemberById(id);
    }

    @PutMapping("/{id}")
    public UserModel updateMember(@RequestBody UserModel userModel, @PathVariable Long id) {
        return userService.updateMember(userModel, id);
    }

    @DeleteMapping("/{id}")
    public String deleteMember(@PathVariable Long id) {
        userService.deleteMember(id);
        return "UserModel deleted successfully";
    }

}

File: src/main/kotlin/com/hyungu/whoseturn/controllers/UserController.kt
package com.hyungu.whoseturn.controllers

import com.hyungu.whoseturn.entities.UserEntity
import com.hyungu.whoseturn.repositories.UserRepository
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*
import java.util.Optional

@RestController
@RequestMapping("/users")
class UserCrudController(private val userRepository: UserRepository) {

    // Get
    @GetMapping
    fun getAllUsers(): Iterable<UserEntity> = userRepository.findAll()

    @GetMapping("/{id}")
    fun getUserById(@PathVariable id: String): Optional<UserEntity> {
        return userRepository.findById(id);
    }

    // Post
    @PostMapping
    fun postUser(@RequestBody user: UserEntity): UserEntity = userRepository.save(user)

    // Put
    @PutMapping("/{id}")
    fun putUser(@PathVariable id: String, @RequestBody user: UserEntity): ResponseEntity<UserEntity> =
        if (userRepository.existsById(id))
            ResponseEntity<UserEntity>(
                userRepository.save(user),
                HttpStatus.OK
            )
        else
            ResponseEntity<UserEntity>(
                userRepository.save(user),
                HttpStatus.CREATED
            )

    // Delete
    @DeleteMapping("/{id}")
    fun deleteUser(@PathVariable id: String) = userRepository.deleteById(id)

}
